src/requestManager.cpp

```diff
@@ -1097,7 +1094,14 @@ bool CRequestManager::MarkBlockAsReceived(const uint256 &hash, CNode *pnode)
                     nIterations++;
                 if (nIterations > nOverallRange)
                 {
-                    nOverallAverageResponseTime -= (nOverallAverageResponseTime / nOverallRange);
+                    if (nOverallRange > 0)
+                    {
+                        nOverallAverageResponseTime -= (nOverallAverageResponseTime / nOverallRange);
+                    }
+                    else
+                    {
+                        LOG(IBD, "Calculation of average response time failed due to division by zero.\n");
+                    }
                 }
                 nOverallAverageResponseTime += nResponseTime / nOverallRange;
```

Comment:
here's another potential / 0.  maybe wrap the entire code in one if check?

Reply:
Good catch, have done so now. Accidentally clicked 'cancel my review' button, I don't know what that did. All/most my comments still seem to be here?